### ISSUE TITLE
Cannot receive the message in mobile side

### ISSUE THREAD

**Issue by tngan** [InterestingWalrus/matrice_autonomy#1]
Hi @InterestingWalrus, I am also working on the sample for exchanging the data in between mobile and onboard sdk through data transparent transmission mentioned in DJI's documentation. have you tried to receive the data in mobile ?

I have found a [doc](https://developer.dji.com/onboard-sdk/documentation/appendix/mobile-onboard-APIs.html) to implement the delegate, but it doesn't work though when I directly call the service with ros command. 

```console
$ rosservice call /dji_sdk/send_data_to_mobile "data: '123456'"
```

**Comment by ybabs** [comment]
@tngan  Yeah I managed to send data to my mobile device. Unfortunately, I'm currently on holiday an d I didn't push the latest version to Github. I don't call ros services directly. I start the ros service within my own implementation based on this sample provided by DJI [here](https://github.com/dji-sdk/Onboard-SDK-ROS/blob/3.6/dji_sdk_demo/src/demo_mobile_comm.cpp).  I then recieve data using a background thread and calling `setOnboardSDKDeviceDataCallback` in the doc you linked above

**Comment by tngan** [comment]
@InterestingWalrus Thanks for your quick reply. Is it possible to send arbitrary data instead of those predefined actions (e.g. landing, takeoff, etc) in your implementation ?

For data transparent transmission, it mentioned the CMD set is 0x00 and id is 0xFE [here](https://developer.dji.com/onboard-sdk/documentation/introduction/index.html#cmd-set-cmd-id), but I don't know where to define the CMD set.

Updates:

I think we can set it when we define the ServiceAck [here](https://github.com/dji-sdk/Onboard-SDK-ROS/blob/1837aae121b1c0aa959eae2f0ad8f14b3b518ed0/dji_sdk_demo/include/dji_sdk_demo/demo_mobile_comm.h#L54-L62). So It's possible to transmit arbitrary data, it's worth to try.

**Comment by ybabs** [comment]
You can send bytes using SendDatatoOnboardSDKDevice.

**Comment by tngan** [comment]
@InterestingWalrus `SendDatatoOnboardSDKDevice` is for upstream from mobile to OES. However, I am working on the downstream from OES back to mobile, the example code of the ROS service call should be something like:

```c++
dji_sdk::SendMobileData mobile_data;
CmdToMobile cmdMobile;
cmdMobile.cmdSet = 0;  // 0x00
cmdMobile.cmdID = 254; // 0xFE for data transparent transmission
mobile_data.request.data.resize(sizeof(CmdToMobile));
memcpy(&mobile_data.request.data[0], (uint8_t *)(&cmdMobile), sizeof(CmdToMobile));
mobile_data_service.call(mobile_data);
```

CmdToMobile is defined as

```c++
typedef struct CmdToMobile{
  uint8_t cmdSet; // 1 byte
  uint8_t cmdID; // 1 byte
  uint8_t[] cmdVal; // should be less than 100 bytes
} CmdToMobile
```

Does it make sense ?

[FYI](https://developer.dji.com/onboard-sdk/documentation/introduction/data-transparent-transmission.html#usage-scenario-1-downstream-from-oes-to-mobile-device)

**Comment by ybabs** [comment]
Yeah that looks about right.

**Comment by tngan** [comment]
@InterestingWalrus Cool, will give it a try today.

**Comment by ybabs** [comment]
@tngan , did you manage to fix your issue, I updated to the latest version of the DJI SDK and I can't also seem to receive data on a mobile device.
